packages/twenty-front/src/modules/workflow/components/WorkflowDiagramStepNode.tsx

```diff
@@ -1,87 +1,51 @@
+import { useTheme } from '@emotion/react';
 import styled from '@emotion/styled';
-import { Handle, Position } from '@xyflow/react';
+import { IconCode, IconPlaylistAdd } from 'twenty-ui';
 
-const StyledStepNodeContainer = styled.div`
+const StyledStepNodeLabelIconContainer = styled.div`
+  align-items: center;
+  background: ${({ theme }) => theme.background.transparent.light};
+  border-radius: ${({ theme }) => theme.spacing(1)};
   display: flex;
-  flex-direction: column;
-
-  padding-bottom: 12px;
-  padding-top: 6px;
-`;
-
-const StyledStepNodeType = styled.div`
-  background-color: ${({ theme }) => theme.background.tertiary};
-  border-radius: ${({ theme }) => theme.border.radius.sm}
-    ${({ theme }) => theme.border.radius.sm} 0 0;
-
-  color: ${({ theme }) => theme.color.gray50};
-  font-size: ${({ theme }) => theme.font.size.xs};
-  font-weight: ${({ theme }) => theme.font.weight.semiBold};
-
-  padding: ${({ theme }) => theme.spacing(1)} ${({ theme }) => theme.spacing(2)};
-  position: absolute;
-  top: 0;
-  transform: translateY(-100%);
-
-  .selectable.selected &,
-  .selectable:focus &,
-  .selectable:focus-visible & {
-    background-color: ${({ theme }) => theme.color.blue};
-    color: ${({ theme }) => theme.font.color.inverted};
-  }
-`;
-
-const StyledStepNodeInnerContainer = styled.div`
-  background-color: ${({ theme }) => theme.background.secondary};
-  border: 1px solid ${({ theme }) => theme.border.color.medium};
-  border-radius: ${({ theme }) => theme.border.radius.md};
-  display: flex;
-  gap: ${({ theme }) => theme.spacing(2)};
-  padding: ${({ theme }) => theme.spacing(2)};
-
-  position: relative;
-  box-shadow: ${({ theme }) => theme.boxShadow.superHeavy};
-
-  .selectable.selected &,
-  .selectable:focus &,
-  .selectable:focus-visible & {
-    background-color: ${({ theme }) => theme.color.blue10};
-    border-color: ${({ theme }) => theme.color.blue};
-  }
-`;
-
-const StyledStepNodeLabel = styled.div`
-  font-size: ${({ theme }) => theme.font.size.md};
-  font-weight: ${({ theme }) => theme.font.weight.medium};
-`;
-
-const StyledSourceHandle = styled(Handle)`
-  background-color: ${({ theme }) => theme.color.gray50};
-`;
-
-export const StyledTargetHandle = styled(Handle)`
-  visibility: hidden;
+  justify-content: center;
+  padding: ${({ theme }) => theme.spacing(1)};
 `;
 
 export const WorkflowDiagramStepNode = ({
   data,
 }: {
   data: WorkflowDiagramStepNodeData;
 }) => {
-  return (
-    <StyledStepNodeContainer>
-      {data.nodeType !== 'trigger' ? (
-        <StyledTargetHandle type="target" position={Position.Top} />
-      ) : null}
-
-      <StyledStepNodeInnerContainer>
-        <StyledStepNodeType>{data.nodeType}</StyledStepNodeType>
+  const theme = useTheme();
+
+  const renderStepIcon = () => {
+    switch (data.nodeType) {
+      case 'trigger': {
+        return (
+          <StyledStepNodeLabelIconContainer>
+            <IconPlaylistAdd size={16} color={theme.font.color.tertiary} />
```

Comment:
size should use the theme

Reply:
Done. Thank you, I didn't know icon sizes where defined in the theme.